### ISSUE TITLE
Event specifications shouldn't include "period" from json at least by default

### ISSUE THREAD

**Issue by travisdowns** [andikleen/pmu-tools#141]
When you use `walk_events`, the `event` spec string you get includes the period field for most/many events. For example `l2_lines_in.all` has the event string `cpu/umask=0x1F,period=100003,event=0xf1`

What's 100003 doing in there? It should be up to the user what period they want, or if they want a period at all (e.g., I just want to use these events in "counter" mode). Since the most common use of the event string is probably to pass it back into `jevent_name_to_attr` or `resolve_event`, both of which respect the `period` attribute, you'll often unintentionally end up setting the period when doing a `perf_event_open`.

The real "bug", if any, is that the 01.org downloads include the SampleAfter value at all, but I guess it is there because these are the default values used by VTune for its various investigations.

I recommend leaving the period out of the json-parsed events returned by `walk_event`, at least by default. If you agree, I can submit a pull request to this effect.

**Comment by andikleen** [comment]
Including period was kind of intentional, inherited from ocperf, to discourage the perf frequency mode which is otherwise used, and which has various issues. But it can be annoying when it overrides an existing period. For counting it should be normally ignored. 
In Linux perf which had a similar problem, i added a weak field for this, but that would be difficult here. 
I guess for now could just not set the period field in the parser when it is already set?

**Comment by travisdowns** [comment]
Maybe I'm mistaken, but doesn't `perf_event_open` key off the period field to determine whether to set up sample/interrupt based mode in the first place? So `period == 0` is special? That field is a `union` in the event_attr structure and the exact semantics are clear to me.

> In Linux perf which had a similar problem, i added a weak field for this, but that would be difficult here.
> I guess for now could just not set the period field in the parser when it is already set?

I didn't quite follow that: in this case the events are being parsed out of the json and returned as a big list (e.g., via the callback in `walk_event`), so there no existing event (at least in my scenario) to examine.

It's not a big issue for me since I can just "zero out" the event, just seemed weird to have all of these random period figures (and not for all events) in the json.

**Comment by andikleen** [comment]
Yes you're right. For counting it should be zeroed. The counting
will work either way, but you would have extra overhead from the PMIs.
perf actually always sets up an PMI, but it's only on 47bit overhflow.

I guess should add something to the documentation. Or maybe invent a new
flag that avoids it.
